.github/workflows/go.yml

```diff
@@ -57,6 +59,26 @@ jobs:
       with:
         version: v1.51.2
 
+  coverage:
+    name: coverage
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+    - uses: actions/checkout@v3
+    - name: Download coverage reports
+      uses: actions/download-artifact@v3
+      with:
+        name: reports
+        path: reports
+
+    - name: Codecov
+      uses: codecov/codecov-action@v3
+      with:
+        directory: reports
+        flags: unittests
+
   release-build:
     name: release-build
```

Comment:
This job should have:
```
permissions:
  contents: write
```
per https://github.com/goreleaser/goreleaser-action#workflow

Reply:
I didn't change the permissions for that job in this PR; when it runs via `push` trigger it will have write access in all the scopes. If we want to pare down permissions for this job to only `contents: write` (it's a good idea!) we should do it in a different PR.

Reply:
one other thing is that we can't actually give it `contents: write` for the `pull_request`-triggered runs if the PR is based on a fork, so the job will actually fail in that case... I don't know what goreleaser is actually doing during PR runs, I guess it's just ensuring it can build and such since it doesn't do an actual release, and in that case it won't need write.